server.js

```diff
@@ -25,11 +73,68 @@ app.use(cors());
 app.use(express.json());
 
 // Start defining your routes here
-app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+app.get('/', (req, res) => {
+  res.json(listEndpoints(app));
+});
+
+// https://project-mongo-api-kpnlmcrmoq-lz.a.run.app/titles
+// http://localhost:8080/titles
+// Route to get all titles
+app.get('/titles', async (req, res) => {
+  const titles = await Title.find().populate('director');
+  res.json(titles)
+});
+
+// https://project-mongo-api-kpnlmcrmoq-lz.a.run.app/titles/645cd880f1686cd98b080ddd
+// http://localhost:8080/titles/645cd880f1686cd98b080ddd
+// Route to get a single title by ID
+app.get('/titles/:id', async (req, res) => {
+  try {
+  const title = await Title.findById(req.params.id).populate('director'); 
+  if (title) {
+    res.json(title)
+  } else {
+    res.status(404).json({ error: 'Title not found '});
+  }
+} catch (error) {
+  console.error(error);
+  res.status(500).json({ error: 'Server error' })
+}
+});
+
+// https://project-mongo-api-kpnlmcrmoq-lz.a.run.app/directors
```

Comment:
I like that you have all the endpoint addresses so it's easy to follow along your code!